spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/domain/entities/Application.java

```diff
@@ -44,8 +50,28 @@
 
 	@lombok.Builder(builderClassName = "Builder", toBuilder = true)
 	private Application(String name, @Nullable BuildVersion buildVersion, @Nullable String status,
-			@Nullable Instant statusTimestamp, List<Instance> instances) {
+			@Nullable Instant statusTimestamp, List<Instance> instances, String groupName) {
+		Assert.notNull(name, "'name' must not be null");
+		Assert.hasText(groupName, "groupName must not be null");
+		Assert.isTrue(CollectionUtils.isEmpty(instances), "instances must not be null or empty");
```

Comment:
Shouldn't you check false? and there is a statement `instances.isEmpty()` at 61 line.  Don't you just have to check for only null?

Reply:
Yeah, you are right.